### PR TITLE
Upcoming Events file sync with WPCOM.

### PR DESCRIPTION
**Pull request by georgestephanis**
#### Changes proposed in this Pull Request:

Syncs up files regarding the Upcoming Events widget and shortcode from WPCOM including some changes by @sirbrillig regarding timezones.

As this is only now finally shipping, shouldn't need a changelog entry.

### CODE REVIEW

File: modules/shortcodes/upcoming-events.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+		if ( ! $events ) {
+			$events = sprintf( '<p>%s</p>', __( 'No upcoming events', 'jetpack' ) );
+		}
+
+		return $events;
+	}
+}
+
+add_action( 'plugins_loaded', array( 'Upcoming_Events_Shortcode', 'init' ), 100 );
```
Comment (dereksmart) [inline modules/shortcodes/upcoming-events.php]: I had to bump this to priority `101` to get it to register.

**Comment by dereksmart** [comment]
Widget changes look good 👍

**Comment by georgestephanis** [comment]
@dereksmart I think I've got what you need, can you give it a skim and confirm?

**Comment by dereksmart** [comment]
@georgestephanis changes look great, thanks!  This is good to ship

**Comment by dereksmart** [comment]
Merged to 4.5 78ee704

**Comment by jordwest** [comment]
We've been seeing some issues on WPCOM with the Upcoming Events widget around daylight savings boundaries. I'm not sure if those issues were pre-existing, but it's possible the issue will come up in Jetpack with this sync. It's a subtle bug that only happens as we approach a daylight savings boundary, when events are scheduled after that boundary. Details: p2UL9c-2Zq-p2

**Pull request merged**